### PR TITLE
Fix mito QC stats JSON conversion crash for bam started-runs when BAM-derived sample id deviates from rerun sample id

### LINKED ISSUE #169: Starting from bam with modified group/sample ids fails at `build_mitochondrial_qc_json`
```
Error executing process > 'build_mitochondrial_qc_json (SampleA-trio-rank-model-fix)'

Caused by:
  Process `build_mitochondrial_qc_json (SampleA-trio-rank-model-fix)` terminated with an error exit status (1)

Command executed:

  mito_tsv_to_json.py SampleA-trio-rank-model-fix_mito_coverage.tsv SampleA-trio-rank-model-fix > "SampleA-trio-rank-model-fix_mito_qc.json"

Command exit status:
  1

Command output:
  (empty)

Command error:
  Traceback (most recent call last):
    File "/fs1/alkc/proj/nextflow_wgs/bin/mito_tsv_to_json.py", line 24, in <module>
      average_coverage = float(row[f"{args.sample_id}_mean_cvg"])
  KeyError: 'SampleA-trio-rank-model-fix_mean_cvg'

Work dir:
 ...

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`
```

In this case the `KeyError` occurred due to the identifier targetted by the python-script grabbing the sample name from the bam filename, which did not align with the sample name used in the rerun.

i.e. the sample data for the sample exists in the TSV but uses the name of the original bam file as sample identifier instead of the sample id specified in the input CSV

### PR DESCRIPTION
**Pull request by alkc**
# Description and reviewer info

Update to process and python script that converts mito sentieon coverage TSV to JSON.

The script previously required sample-id input to properly access specific fields in the TSV. 
This has now been changed so that the same values are accessed by suffix, under the strong 
assumption that the TSV contains data from only one sample.

Fixes #169 

## Type of change

- [ ] Documentation
- [ ] Patch
- [x] Minor change
- [ ] Major change 

# Checklist

- [x] Self-review of my code
- [x] Update the CHANGELOG
- [ ] Tag the latest commit (vX.Y.Z format)
- [ ] Log samples used for testing in the `Verification_samples_log` Excel sheet

## Minor change / patch-ish
- [ ] bam-started wgs run completes with custom sample id and mito QC data checks out
- [x] wgs trio run completes started from bjorn/fastq and mito QC data checks out for all samples
- [x] Updated mito script fetches the correct values and outputs correct JSON to stdout
- [x] At least one other person has reviewed and approved my code
- [x] I have made corresponding changes to the documentation (software versions, etc.)

# Test/review documentation

## Review performed by

- [ ] Alexander
- [x] Jakob
- [ ] Paul
- [ ] Ryan
- [ ] Viktor

(Add if missing)

## Testing performed by

- [x] Alexander
- [ ] Jakob
- [ ] Paul
- [ ] Ryan
- [ ] Viktor

### CODE REVIEW

File: bin/mito_tsv_to_json.py [hunk 1]
```diff
@@ -1,29 +1,58 @@
...
 import sys
 
+AVERAGE_COVERAGE_FIELD_SUFFIX = "_mean_cvg"
+PCT_ABOVE_500X_FIELD_SUFFIX = "_%_above_500"
+
 # Parse command-line arguments
 parser = argparse.ArgumentParser(description="Convert Mito coverage TSV file to JSON")
 parser.add_argument("mito_coverage_tsv", help="Path to the input TSV file")
-parser.add_argument("sample_id", help="Sample ID to extract data for")
+parser.add_argument("--sample_id", help="Sample ID to extract data for", required=False)
```
Comment (alkc) [inline bin/mito_tsv_to_json.py:18]:
```suggestion
```

Forgot to remove this.

File: bin/mito_tsv_to_json.py [hunk 2]
```diff
@@ -1,29 +1,58 @@
...
+    The fields are prefixed with a sample-name derived from the BAM-file,
+    which can deviate from the group/sample ids specified in the input CSV
+    e.g. when the pipeline is started from an old BAM but with new sample ids
+    specified in the CSV.
+
+    The TSVs are generated per-sample, meaning there is no risk of the same
+    TSV containing multiple samples, which would require a more precise
+    dict access.
+    """
+    for key, value in row.items():
```
[the remark this answers is not in the document]
Comment (Jakob37) [reply bin/mito_tsv_to_json.py:35]: Maybe less critical, but you could do the same for the number of rows. You are assuming a format now. Like my wife's former Ph.D. supervisor liked to say. Trust is good. Double check is better.

**Review by Jakob37 — APPROVED** [review]
Looks good!

Just some small reflections.

File: bin/mito_tsv_to_json.py [hunk 3]
```diff
@@ -1,29 +1,58 @@
...
+
 # Parse command-line arguments
 parser = argparse.ArgumentParser(description="Convert Mito coverage TSV file to JSON")
 parser.add_argument("mito_coverage_tsv", help="Path to the input TSV file")
-parser.add_argument("sample_id", help="Sample ID to extract data for")
+parser.add_argument("--sample_id", help="Sample ID to extract data for", required=False)
 args = parser.parse_args()
 
+
+def extract_value_by_suffix(row: dict, key_suffix: str) -> str:
```
[the remark this answers is not in the document]
Comment (alkc) [reply bin/mito_tsv_to_json.py:22]: nope. python3 in the wgs container is too old for that. 

`typing.Dict` might work though. i'll check.

File: bin/mito_tsv_to_json.py [hunk 4]
```diff
@@ -1,29 +1,58 @@
 #!/usr/bin/env python3
 
 # Author:      Alexander Koc <alexander.koc@skane.se>
-# Date:        2023-06-21
 # Description: Take TSV output generated by "process sentieon_mitochondrial_qc"
```
[the remark this answers is not in the document]
Comment (alkc) [reply bin/mito_tsv_to_json.py:4]: Good idea

File: bin/mito_tsv_to_json.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alkc) [reply bin/mito_tsv_to_json.py]: Fixed!

File: bin/mito_tsv_to_json.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jakob37) [reply bin/mito_tsv_to_json.py]: You could try this:

https://peps.python.org/pep-0585/

I.e. importing 

```
from __future__ import annotations
```

And then

```
dict[str, str]
```

Not sure it works. But curious to know

**Comment by alkc** [comment]
Scrpt confirmed to work for commit [844ae20](https://github.com/Clinical-Genomics-Lund/nextflow_wgs/pull/232/commits/844ae20532b757cfb2c6c80872b9da6dee345314) in wgs trio.

**Comment by alkc** [comment]
I'm omitting scout testing here, as this part of the pipeline does not affect scout loads.

I've confirmed that the QC data for CDM is a merged parseable json for both wgs single and trio.

**Comment by alkc** [comment]
The update resolves one particular obstacle in bam-started pipeline runs, but there is work remaining that will have to be resolved in a separate patch: #233

**Pull request merged**